pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -171,6 +178,31 @@ func (o *MustGatherOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, arg
 	if len(o.DestDir) == 0 {
 		o.DestDir = fmt.Sprintf("must-gather.local.%06d", rand.Int63())
 	}
+	// TODO: this should be in Validate() method, but added here because of the call to o.completeImages() below
+	if o.AllImages {
+		errStr := fmt.Sprintf("and --all-images are mutually exclusive: please specify one or the other")
+		if len(o.Images) != 0 {
+			return fmt.Errorf("--image %s", errStr)
+		}
+		if o.HostNetwork {
+			return fmt.Errorf("--host-network %s", errStr)
+		}
+		if len(o.ImageStreams) != 0 {
+			return fmt.Errorf("--image-streams %s", errStr)
+		}
+		if o.NodeName != "" {
+			return fmt.Errorf("--node-name %s", errStr)
+		}
+		if o.RunNamespace != "" {
+			return fmt.Errorf("--run-namespace %s", errStr)
+		}
+		if o.SourceDir != "/must-gather/" {
```

Comment:
Why we don't allow changing the source directory?

Reply:
Discussion happened here - https://github.com/openshift/enhancements/pull/1487#discussion_r1402232967. 

If you recommend letting users specify this particular option, I can do so.

Reply:
> Why we don't allow changing the source directory?

Based on its help message, I suppose it's not possible to specify a source dir for multiple images? So using it in conjunction with `--all-images` will look for it on all the Pods?

I'm not 100% sure. If you recommend, I'd remove it.

Reply:
IIUC, `--source-dir` flag shouldn't be specified with `--all-images` flag, right?

Reply:
https://github.com/openshift/enhancements/pull/1487#discussion_r1402232967 does not forbid that. The statement in https://github.com/openshift/enhancements/pull/1487/files#diff-890eb7a924799156fec30313ddd05cb387f5a9255eedbe00f899639bcff70700R128-R129 is to restrictive wrt. --source-dir. Specifying `--source-dir` with --all-images is fine. Worth updating the enhancement text to allow the flag as well.

Reply:
Done